hazelcast/src/main/java/com/hazelcast/internal/adapter/DataStructureAdapterMethod.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.adapter;
+
+/**
+ * Defines the signature of a method in {@link DataStructureAdapter} to search for available methods.
+ */
+public interface DataStructureAdapterMethod {
```

Comment:
I pulled out this interface so I can test the `MethodAvailableMatcher`  without adding garbage to the real code.